src/v/cloud_storage/remote_segment.cc

```diff
@@ -315,18 +315,18 @@ remote_segment::offset_data_stream(
     offset_index::find_result pos;
     std::optional<uint16_t> prefetch_override = std::nullopt;
     if (first_timestamp) {
-        // Time queries are linear search from front of the segment.  The
-        // dominant cost of a time query on a remote partition is promoting
-        // the segment into our local cache: once it's here, the cost of
-        // a scan is comparatively small.  For workloads that do many time
-        // queries in close proximity on the same partition, an additional
-        // index could be added here, for hydrated segments.
-        pos = {
-          .rp_offset = _base_rp_offset,
-          .kaf_offset = _base_rp_offset - _base_offset_delta,
-          .file_pos = 0,
-        };
+        // We look up the timestamp in the index to reduce the number of chunks
+        // which will require hydration. The dominant cost of a time query on a
+        // remote partition is promoting the chunks into our local cache: once
+        // they're here, the cost of a scan is comparatively small.
+
         prefetch_override = 0;
+        pos = maybe_get_offsets(*first_timestamp)
+                .value_or(offset_index::find_result{
```

Comment:
Nit: should the `value_or` handling be moved outside the `if`? Both branches now handle a failed offset lookup with the same default value, so it can remove some duplication.

Reply:
Good suggestion. Done.